editor/progress_dialog.cpp

```diff
@@ -147,6 +157,12 @@ void ProgressDialog::_popup() {
 		}
 		// No host window found, use main window.
 		EditorInterface::get_singleton()->popup_dialog_centered(this, ms);
+	} else {
+		Rect2i adjust = _popup_adjust_rect();
+		if (adjust != Rect2i()) {
+			set_position(adjust.position);
+			set_size(adjust.size);
+		}
```

Comment:
I would invert the condition for better readability (to prevent a double negative).